worker/internal/benthos/transformers/utils/string_utils.go

```diff
@@ -121,3 +122,23 @@ func IsValidUsername(username string) bool {
 
 	return matched
 }
+
+// use MaxASCII to ensure that the unicode value is only within the ASCII block which only contains latin numbers, letters and characters
+func IsValidChar(s string) bool {
+	for _, r := range s {
+		if !(r <= unicode.MaxASCII && (unicode.IsNumber(r) || unicode.IsLetter(r) || unicode.IsSpace(r) || IsAllowedSpecialChar(r))) {
+			return false
+		}
+	}
+	return true
+}
+
+func IsAllowedSpecialChar(r rune) bool {
+	allowedSpecialChars := "!@#$%^&*()-+=_[]{}|\\ ;\"<>,./?"
+	for _, allowedChar := range allowedSpecialChars {
+		if r == allowedChar {
+			return true
+		}
+	}
+	return false
```

Comment:
this can be made much faster over time if you turn this into a set.
```
var allowedSpecialChars = map[rune]any{"!": struct{}{}, ...}
func IsAllowedSpecialChar(r rune) bool {
  _, ok := allowedSpecialChars[r]
 return ok
}
```

Reply:
updated this by creating a map of the allowed special characters and using to check if a character is is a specal character or not